src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/GenericLookupResult.cs

```diff
@@ -147,6 +152,39 @@ public override void AppendMangledName(NameMangler nameMangler, Utf8StringBuilde
     }
 
     /// <summary>
+    /// Generic lookup result that points to a virtual function address load stub.
+    /// </summary>
+    internal sealed class VirtualResolveGenericLookupResult : GenericLookupResult
+    {
+        private MethodDesc _method;
+
+        public VirtualResolveGenericLookupResult(MethodDesc method)
+        {
+            Debug.Assert(method.IsRuntimeDeterminedExactMethod);
+            Debug.Assert(method.IsVirtual);
+
+            // Normal virtual methods don't need a generic lookup.
+            Debug.Assert(method.OwningType.IsInterface || method.HasInstantiation);
+
+            _method = method;
+        }
+
+        public override ISymbolNode GetTarget(NodeFactory factory, Instantiation typeInstantiation, Instantiation methodInstantiation)
+        {
+            MethodDesc instantiatedMethod = _method.InstantiateSignature(typeInstantiation, methodInstantiation);
+            return factory.ReadyToRunHelper(ReadyToRunHelperId.VirtualCall, instantiatedMethod);
```

Comment:
Put in a comment describing why this shouldn't be here... Something like, the function pointer returned should be the result of the virtual dispatch, not a pointer to the virtual dispatcher.